reporting/api/src/main/java/org/jboss/windup/reporting/config/HasHint.java

```diff
@@ -34,51 +35,31 @@ public boolean evaluate(GraphRewrite event, EvaluationContext context, WindupVer
         ExecutionStatistics.get().begin(HasHint.class.getCanonicalName());
         try
         {
-            boolean result = false;
             InlineHintService service = new InlineHintService(event.getGraphContext());
 
-            if (payload instanceof FileReferenceModel)
-            {
-                Iterable<InlineHintModel> hints = service.getHintsForFileReference((FileReferenceModel) payload);
-                if (messagePattern == null)
-                {
-                    result = hints.iterator().hasNext();
-                }
-                else
-                {
-                    for (InlineHintModel c : hints)
-                    {
-                        ParameterizedPatternResult parseResult = messagePattern.parse(c.getHint());
-                        if (parseResult.matches() && parseResult.isValid(event, context))
-                        {
-                            result = true;
-                            break;
-                        }
-                    }
-                }
-            }
+            Iterable<InlineHintModel> hints = null;
+            // FileLocationModel is a fake FileReferenceModel with [1,1] row and column, so we want hints for the whole file.
```

Comment:
(16:56:00) ozizka-FN: jsightler:  See the PR ^
(16:56:19) ozizka-FN: Maybe I'm still following some bad logic. Looking at it
(16:56:38) ozizka-FN: Now I have to go pick up Hanka, but will look. Or tell me that it's really non-sense :)
(16:56:53) jsightler: Yeah, that logic is basically broken.
(16:59:18) ozizka-FN: But it seems to work, and also work for all tests, so far (still running).
(16:59:21) jsightler: Basically now if I actually do care about only hints that come file File.exists that will no longer be possible.
(16:59:40) jsightler: Its trading one set of limited semantics for a different set of limited semantics.
(17:00:05) ozizka-FN: jsightler:  Could you write a XML for that?
(17:00:11) jsightler: It would be far better to just make has-hint configurable.
(17:00:20) ozizka-FN: Hmm perhaps
(17:00:59) jsightler: (also, it would be better if rules like this were written in a rule language instead of the simplified xml syntax)
(17:01:01) ozizka-FN: <has-hint at="whole-file|line"> ?
(17:01:15) jsightler: Yeah, something like that.
(17:01:25) ozizka-FN: Well, that's completely different thing :)  Windup 4
(17:01:48) jsightler: Actually that isn't bad at all... that sort of allows it to generalize to cases where the condition isn't a File condition